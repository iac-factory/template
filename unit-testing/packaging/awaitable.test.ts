import { initialize } from ".";

describe( "awaitable", () => {
    initialize();

    it( "import", async () => {
        expect.assertions(1);

        const main = import("..");
        const snapshot = JSON.stringify(main, null, 4);
        const result = "Successful";

        const state: import("Unit-Testing").State = {
            ... expect.getState(), ... {
                data: { result, snapshot }
            }
        };

        await expect( main ).resolves.toBeTruthy().then(() => {
            expect.setState( state );

            expect(snapshot).toMatchSnapshot();
        });
    } );
} );
