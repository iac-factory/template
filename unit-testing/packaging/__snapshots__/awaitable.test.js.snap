// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Awaitable Import 1`] = `
"{
    \\"Package\\": \\"[Redacted]\\",
    \\"CLI\\": {},
    \\"TLS\\": {},
    \\"Reader\\": {},
    \\"C\\": {},
    \\"Filters\\": {},
    \\"Color\\": {
        \\"bright\\": {},
        \\"Names\\": {
            \\"black\\": \\"black\\",
            \\"red\\": \\"red\\",
            \\"green\\": \\"green\\",
            \\"yellow\\": \\"yellow\\",
            \\"blue\\": \\"blue\\",
            \\"magenta\\": \\"magenta\\",
            \\"cyan\\": \\"cyan\\",
            \\"white\\": \\"white\\",
            \\"gray\\": \\"gray\\",
            \\"grey\\": \\"grey\\"
        }
    },
    \\"Modifiers\\": {},
    \\"Interface\\": {
        \\"enabled\\": true,
        \\"visible\\": true
    }
}"
`;
